pyccel/codegen/utilities.py

```diff
@@ -51,19 +51,19 @@
     "cwrapper"        : ("cwrapper", CompileObj("cwrapper.c",folder="cwrapper", accelerators=('python',))),
     "numpy_f90"       : ("numpy", CompileObj("numpy_f90.f90",folder="numpy")),
     "numpy_c"         : ("numpy", CompileObj("numpy_c.c",folder="numpy")),
-    "Set_extensions"  : ("STC_Extensions", CompileObj("Set_Extensions.h",
+    "STC_Extensions/Set_extensions"  : ("STC_Extensions", CompileObj("Set_Extensions.h",
                                                       folder="STC_Extensions",
```

Comment:
I don't think the folder is needed if you use `STC_Extensions/List_Extension.h`. Ideally the enclosing folder should be specified but I don't know if there's a way to do this?

```suggestion
                                                       folder=".",
```
?

Reply:
```Python
"gFTL_functions/Set_extensions"  : ("gFTL_functions", CompileObj("Set_Extensions.inc",
                                                                     folder="gFTL_functions",
                                                                     has_target_file = False,
                                                                     dependencies = (external_libs['gFTL'][1],))),
```
I followed this pattern and I think that your suggestion will create a deadlock because the enclosing folder will be say `__pyccel__` and thus can't be acquired.

Reply:
Ah! Good point. I didn't think about the fact that the folder is used for avoiding deadlocks too. In this case it should indeed be `STC_Extensions` to protect these files